env0/resource_template.go

```diff
@@ -23,7 +23,7 @@ var allowedTemplateTypes = []string{
 	"cloudformation",
 }
 
-func resourceTemplate() *schema.Resource {
+func getTemplateSchema(prefix string) map[string]*schema.Schema {
```

Comment:
Not a big change here (just looks scary).
The schema varies a little bit between the template use-case and an environment-without-template use-case.
In this way, if more fields are added to the template resource, no work is needed.